EventFilter/GEMRawToDigi/plugins/GEMPackingTester.cc

```diff
@@ -0,0 +1,119 @@
+#include "DataFormats/FEDRawData/interface/FEDTrailer.h"
+#include "DataFormats/GEMDigi/interface/GEMDigiCollection.h"
+
+using namespace std;
+class GEMPackingTester : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+public:
+  explicit GEMPackingTester(const edm::ParameterSet&);
+  ~GEMPackingTester() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginJob() override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  edm::EDGetTokenT<FEDRawDataCollection> fedToken_;
+  edm::EDGetTokenT<GEMDigiCollection> gemDigiToken_;
+  edm::EDGetTokenT<GEMDigiCollection> gemSimDigiToken_;
+  bool readMultiBX_;
+
+  TTree* tree_;
+  int b_ge0, b_ge1, b_ge2;
+};
+
+GEMPackingTester::GEMPackingTester(const edm::ParameterSet& iConfig)
+    : fedToken_(consumes<FEDRawDataCollection>(iConfig.getParameter<edm::InputTag>("fed"))),
+      gemDigiToken_(consumes<GEMDigiCollection>(iConfig.getParameter<edm::InputTag>("gemDigi"))),
+      gemSimDigiToken_(consumes<GEMDigiCollection>(iConfig.getParameter<edm::InputTag>("gemSimDigi"))),
+      readMultiBX_(iConfig.getParameter<bool>("readMultiBX")) {
+  usesResource("TFileService");
+  edm::Service<TFileService> fs;
+  tree_ = fs->make<TTree>("fed", "fed");
+  tree_->Branch("ge0", &b_ge0, "ge0/I");
+  tree_->Branch("ge1", &b_ge1, "ge1/I");
+  tree_->Branch("ge2", &b_ge2, "ge2/I");
+}
+
+GEMPackingTester::~GEMPackingTester() {}
+
+void GEMPackingTester::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  b_ge0 = 0;
+  b_ge1 = 0;
+  b_ge2 = 0;
+
+  auto const& fed_buffers = iEvent.get(fedToken_);
+
+  for (unsigned int fedId = FEDNumbering::MINGEMFEDID; fedId <= FEDNumbering::MAXGEMFEDID; ++fedId) {
+    const FEDRawData& fedData = fed_buffers.FEDData(fedId);
+
+    if (fedId == 1473 or fedId == 1474)
+      b_ge0 += fedData.size();
+    if (fedId == 1467 or fedId == 1468)
+      b_ge1 += fedData.size();
+    if (fedId == 1469 or fedId == 1470)
+      b_ge2 += fedData.size();
+  }
+
+  auto const& gemDigis = iEvent.get(gemDigiToken_);
+  auto const& gemSimDigis = iEvent.get(gemSimDigiToken_);
+
+  for (auto const& simDigi : gemSimDigis) {
+    const GEMDetId& gemId = simDigi.first;
+    const GEMDigiCollection::Range& sim = simDigi.second;
+    const GEMDigiCollection::Range& packed = gemDigis.get(gemId);
+
+    for (auto digi = sim.first; digi != sim.second; ++digi) {
+      if (!readMultiBX_ && digi->bx() != 0)
+        continue;
+
+      bool foundDigi = false;
+      for (auto unpackeddigi = packed.first; unpackeddigi != packed.second; ++unpackeddigi) {
+        if ((digi->strip() == unpackeddigi->strip()) && (digi->bx() == unpackeddigi->bx()))
+          foundDigi = true;
+      }
+      if (!foundDigi) {
+        cout << "simMuonGEMDigi NOT found " << gemId << " " << digi->strip() << " " << digi->bx() << endl;
```

Comment:
I am sorry but couts are discourgaed, please use edm::Log or similar instead

Reply:
It was a question on the master PR also with clarification that this is for the test. However, if change will be applied, please propagate to master also.
https://github.com/cms-sw/cmssw/pull/34504/files#r674189118

Thanks.

Reply:
done

Reply:
i will update this in master with another PR soon after #34614 is merged.

Reply:
Thanks @jshlee 
FYI @qliphy @perrotta This backport will not be the same as master, the master will be updated also.

Reply:
(I didn't point out this `cout` in the reco review since this seems to be a testing module not for production, but using MessageLogger is of course preferred)